File: src/StealME.Server/StealME.Server.Messaging/Requests/GetRequest.cs
using StealME.Server.Messaging.Enums;

namespace StealME.Server.Messaging.Requests
{
    public class GetRequest
    {
        public int Id;
        public int GetTypeId;

        public static GetRequest GetStatus()
        {
            return new GetRequest { GetTypeId = (int)GetRequestType.Status};
        }

        public static GetRequest GetIMEI()
        {
            return new GetRequest { GetTypeId = (int)GetRequestType.IMEI };
        }

        public static GetRequest GetProtocolVersion()
        {
            return new GetRequest { GetTypeId = (int)GetRequestType.ProtocolVersion };
        }
    }
}

File: src/StealME.Client/src/stealme/client/communication/Enums/GetRequest.java
package stealme.client.communication.Enums;

public enum GetRequest {
	Status (0), 
	IMEI (1),
	ProtocolVersion(2);
	
	private final int index;   

	GetRequest(int index) {
        this.index = index;
    }

    public int index() { 
        return index; 
    }
}
